Social share

tharain opened the pull request:
used vue_social_share plugin
added child route to be same as parent route?
mainly on chatview.vue

![lint](https://user-images.githubusercontent.com/12406584/30911558-b3f7b204-a3bb-11e7-8191-acc507b57925.png)

tharain committed:
external code from https://github.com/bradvin/social-share-urls

tharain committed:
simple sharing component

tharain committed:
Add assets and refactor Share.vue social platforms

tharain committed:
Increase spacing and remove redundant social media platforms

tharain committed:
change menu logo

tharain committed:
Merge branch 'refs/heads/pr/20' into social_share

# Conflicts:
#	src/store/index.js
#	src/views/QuackAppView.vue

tharain committed:
Fix previous merge reference

tharain committed:
Fix route on sharing button

tharain committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into social_share

# Conflicts:
#	src/store/index.js

tharain committed:
update icons

ngsikai commented on src/router/index.js:
```diff
@@ -27,6 +28,12 @@ const Router = new VueRouter({
     {
       path: '/chat/:roomId',
       name: 'ChatView',
+      children: [
```
May I know what this route is used for?

tharain replied:
at first I wanted to make it /chat/xx/share, but it was buggy (2 share modal appear every-time), so i used it to concatenate the share.vue into the chatView.vue

KnightNiwrem committed:
Merge branch 'social_share' of github.com:nus-chatbox/quack into daphne-social

tharain committed:
removed whatsapp and added meta image to index.html

this will remove not working sharing function and enable user to see the
correct image when sharing

tharain commented:
Added meta tag to the index.html to allow facebook and google to get the correct image. Not sure will it work or not.

KnightNiwrem committed:
Merge branch 'social_share' of github.com:nus-chatbox/quack into daphne-social

KnightNiwrem committed:
Converted sharing modal into component using current page's url as default

KnightNiwrem committed:
Removed telegram after all

tharain committed:
Merge pull request #34 from KnightNiwrem/daphne-social

Fix inability to share current chat url

KnightNiwrem committed:
Use jwt username after all

KnightNiwrem committed:
Merge branch 'master' of github.com:nus-chatbox/quack

KnightNiwrem committed:
Update manifest

tharain committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into social_share

# Conflicts:
#	src/views/ChatView.vue

KnightNiwrem committed:
Bug fix for message patching and fetching last messages

tharain committed:
update chat ui for sharing

KnightNiwrem committed:
Prevent jumpiness when hitting enter on chat

KnightNiwrem committed:
Merge branch 'social_share' of github.com:nus-chatbox/quack

KnightNiwrem committed:
Add relative url ability
